Amplitude/Amplitude.m

```diff
@@ -1313,12 +1331,22 @@ - (void)setEventUploadMaxBatchSize:(int) eventUploadMaxBatchSize
 
 - (BOOL)optOut
 {
+    if (_apiKey == nil) {
+        AMPLITUDE_ERROR(@"ERROR: apiKey cannot be nil or empty, set apiKey with initializeApiKey: before toggling and fetching optOut");
+        return NO;
+    }
     return [[self.dbHelper getLongValue:OPT_OUT] boolValue];
 }
 
 - (void)setDeviceId:(NSString*)deviceId
 {
+    if (_apiKey == nil) {
```

Comment:
This is tricky. This and `optOut` feel like they should be callable before initializing the API since initializeApiKey (some customers probably do this). However, they store data in the database so the database file needs to created before these get called.

It's an API breaking change but what do you think of forcing an API Key when calling instance?